src/main/java/com/icthh/xm/tmf/ms/communication/service/mail/EmailTemplateService.java

```diff
@@ -44,4 +57,29 @@ public RenderTemplateResponse renderEmailContent(RenderTemplateRequest renderTem
             throw new RenderTemplateException(e.getMessage(), renderTemplateRequest.getContent(), renderTemplateRequest.getModel());
         }
     }
+    @SneakyThrows
+    public void updateTemplate(String templateKey, UpdateTemplateRequest updateTemplateRequest) {
+        EmailSpec emailSpec = emailSpecService.getEmailSpec();
+
+        EmailTemplateSpec emailTemplateSpec = emailSpec.getEmails().stream()
+                                                        .filter((spec) -> spec.getTemplateKey().equals(templateKey))
+                                                        .findFirst()
+                                                        .map((spec) -> {
+                                                            spec.setName(updateTemplateRequest.getTemplateName());
+                                                            spec.setSubjectTemplate(updateTemplateRequest.getTemplateSubject());
+                                                            return spec;
+                                                        }).orElseThrow(() -> new EntityNotFoundException("Email template specification not found"));
```

Comment:
Please extract this codebase to method for finding EmailTemplateSpec by key to make it reusable.
For example: 
```
private EmailTemplateSpec getEmailTemplateSpecByKey(String templateKey) {
        return emailSpecService.getEmailSpec().getEmails()
            .stream()
            .filter((spec) -> spec.getTemplateKey().equals(templateKey))
            .findFirst()
            .orElseThrow(() -> new EntityNotFoundException("Email template specification not found"));
    }
`

Reply:
done